src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -412,103 +450,178 @@ private void UpdateCacheIfNeeded(long position, INamedOnnxValueGetter[] srcNamed
+                    var denseTensor = namedOnnxValue.AsTensor<T>() as System.Numerics.Tensors.DenseTensor<T>;
+                    if (denseTensor == null)
+                        throw Host.Except($"Output column {namedOnnxValue.Name} doesn't contain a DenseTensor of expected type {typeof(T)}");
+                    var editor = VBufferEditor.Create(ref dst, (int)denseTensor.Length);
+                    denseTensor.Buffer.Span.CopyTo(editor.Values);
+                    dst = editor.Commit();
+                };
+                return valueGetter;
             }
 
-            private Delegate MakeGetter<T>(DataViewRow input, int iinfo, INamedOnnxValueGetter[] srcNamedValueGetters, string[] activeOutputColNames, OutputCache outputCache)
+            private Delegate MakeStringTensorGetter(DataViewRow input, int iinfo, INamedOnnxValueGetter[] srcNamedValueGetters, string[] activeOutputColNames)
             {
                 Host.AssertValue(input);
-                ValueGetter<VBuffer<T>> valuegetter = (ref VBuffer<T> dst) =>
+                var outputCacher = new OnnxRuntimeOutputCacher();
+                ValueGetter<VBuffer<ReadOnlyMemory<char>>> valueGetter = (ref VBuffer<ReadOnlyMemory<char>> dst) =>
                 {
-                    UpdateCacheIfNeeded(input.Position, srcNamedValueGetters, activeOutputColNames, outputCache);
-                    var namedOnnxValue = outputCache.Outputs[_parent.Outputs[iinfo]];
-                    var denseTensor = namedOnnxValue.AsTensor<T>() as System.Numerics.Tensors.DenseTensor<T>;
+                    UpdateCacheIfNeeded(input.Position, srcNamedValueGetters, activeOutputColNames, outputCacher);
+                    var namedOnnxValue = outputCacher.Outputs[_parent.Outputs[iinfo]];
+                    var denseTensor = namedOnnxValue.AsTensor<string>();
                     if (denseTensor == null)
-                        throw Host.Except($"Output column {namedOnnxValue.Name} doesn't contain a DenseTensor of expected type {typeof(T)}");
+                        throw Host.Except($"Output column {namedOnnxValue.Name} doesn't contain a DenseTensor of expected type {typeof(string)}");
+
+                    // Tensor<string> to Tensor<ReadOnlyMemory<char>>.
+                    var casted = denseTensor.Select(value => value.AsMemory());
+                    // Tensor<ReadOnlyMemory<char>> to Span<ReadOnlyMemory<char>>.
+                    // Note that string vector in ML.NET is typed to Span<ReadOnlyMemory<char>>.
+                    var buffer = new Span<ReadOnlyMemory<char>>(casted.ToArray());
+
+                    // Create VBufferEditor to fill "dst" with the values in "buffer".
                     var editor = VBufferEditor.Create(ref dst, (int)denseTensor.Length);
-                    denseTensor.Buffer.Span.CopyTo(editor.Values);
+                    buffer.CopyTo(editor.Values);
                     dst = editor.Commit();
```

Comment:
>dst = editor.Commit(); [](start = 20, length = 22)

dst= new VBuffer<>

Reply:
No, don’t make a new buffer here. Reuse the dst buffer.

Reply:
You're right.